packages/terra-button/src/Button.jsx

```diff
@@ -115,221 +115,194 @@ const defaultProps = {
   variant: ButtonVariants.NEUTRAL,
 };
 
-class Button extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { active: false, focused: false };
-    this.handleKeyDown = this.handleKeyDown.bind(this);
-    this.handleKeyUp = this.handleKeyUp.bind(this);
-    this.handleOnBlur = this.handleOnBlur.bind(this);
-    this.handleFocus = this.handleFocus.bind(this);
-    this.handleMouseDown = this.handleMouseDown.bind(this);
-    this.handleClick = this.handleClick.bind(this);
-
-    this.shouldShowFocus = true;
-  }
-
-  static getDerivedStateFromProps(nextProps, prevState) {
-    if (prevState.focused && nextProps.isDisabled) {
-      return { focused: false };
+const Button = ({
+  icon,
+  isBlock,
+  isCompact,
+  isDisabled,
+  isIconOnly,
+  isReversed,
+  text,
+  type,
+  variant,
+  href,
+  onClick,
+  onMouseDown,
+  onBlur,
+  onFocus,
+  onKeyDown,
+  onKeyUp,
+  refCallback,
+  title,
+  ...customProps
+}) => {
+  const theme = React.useContext(ThemeContext);
+  const [focused, setFocused] = useState(false);
+  const [active, setActive] = useState(false);
```

Comment:
We should use isFocused and isActive for our getter methods since these are Boolean state variables?